recipes/mimalloc/all/conanfile.py

```diff
@@ -102,12 +102,12 @@ def _configure_cmake(self):
         self._cmake = CMake(self)
         if self._cmake.is_multi_configuration:
             self._cmake.definitions["CMAKE_BUILD_TYPE"] = self.settings.build_type
-        self._cmake.definitions["MI_BUILD_TESTS"] = False
-        self._cmake.definitions["MI_BUILD_SHARED"] = self.options.shared
-        self._cmake.definitions["MI_BUILD_STATIC"] = not self.options.shared and not self.options.get_safe("single_object", False)
-        self._cmake.definitions["MI_BUILD_OBJECT"] = self.options.get_safe("single_object", False)
-        self._cmake.definitions["MI_OVERRIDE"] = self.options.override
-        self._cmake.definitions["MI_SECURE"] = self.options.secure
+        self._cmake.definitions["MI_BUILD_TESTS"] = "OFF"
+        self._cmake.definitions["MI_BUILD_SHARED"] = "ON" if self.options.shared else "OFF"
+        self._cmake.definitions["MI_BUILD_STATIC"] = "ON" if not self.options.shared and not self.options.get_safe("single_object", False) else "OFF"
+        self._cmake.definitions["MI_BUILD_OBJECT"] = "ON" if self.options.get_safe("single_object", False) else "OFF"
+        self._cmake.definitions["MI_OVERRIDE"] = "ON" if self.options.override else "OFF"
+        self._cmake.definitions["MI_SECURE"] = "ON" if self.options.secure else "OFF"
```

Comment:
This should no be needed conan handles this

Reply:
Unfortunately upstream CMake handles some of this options like this:

```cmake
if (MI_BUILD_TESTS MATCHES "ON")
#...
endif()
```

I thought all of them were been dealt with using the `MATCHES` operator, but actually the ones being dealt with so are only: `MI_BUILD_TESTS`, `MI_OVERRIDE` and `MI_SECURE`...

Reply:
Wow, you're right @tarc.
That's some *robust code* by Microsoft 🙄 .

Reply:
Well, I felt guilty and posted a PR for them: https://github.com/microsoft/mimalloc/pull/349

Reply:
Sorry, I forgot who upstream was! You are correct